katsdptelstate/aio/test/test_telescope_state.py

```diff
@@ -31,520 +33,512 @@
-
-    async def test_add_duplicate(self) -> None:
-        await self.ts.add('test_key', 'value', 1234.5)
-        await self.ts.add('test_key', 'value', 1234.5)
-        self.assertEqual([('value', 1234.5)], await self.ts.get_range('test_key', st=0))
-
-    async def test_wait_key_already_done_mutable(self) -> None:
+        assert [(8192, 1), (16384, 2)] == await ts.get_range('test_key', et=2, include_end=True)
+        assert [(8192, 1), (16384, 2), (4096, 3)] == \
+            await ts.get_range('test_key', st=2, et=3, include_previous=True, include_end=True)
+
+    async def test_add_duplicate(self, ts: TelescopeState) -> None:
+        await ts.add('test_key', 'value', 1234.5)
+        await ts.add('test_key', 'value', 1234.5)
+        assert [('value', 1234.5)] == await ts.get_range('test_key', st=0)
+
+    async def test_wait_key_already_done_mutable(self, ts: TelescopeState) -> None:
         """Calling wait_key with a condition that is met must return (mutable version)."""
-        await self.ts.add('test_key', 123)
-        value, timestamp = (await self.ts.get_range('test_key'))[0]
-        await self.ts.wait_key('test_key', lambda v, t: v == value and t == timestamp)
+        await ts.add('test_key', 123)
+        value, timestamp = (await ts.get_range('test_key'))[0]
+        await ts.wait_key('test_key', lambda v, t: v == value and t == timestamp)
 
-    async def test_wait_key_already_done_immutable(self) -> None:
+    async def test_wait_key_already_done_immutable(self, ts: TelescopeState) -> None:
         """Calling wait_key with a condition that is met must return (immutable version)."""
-        await self.ts.add('test_key', 123, immutable=True)
-        await self.ts.wait_key('test_key', lambda v, t: v == 123 and t is None)
+        await ts.add('test_key', 123, immutable=True)
+        await ts.wait_key('test_key', lambda v, t: v == 123 and t is None)
 
-    async def test_wait_key_already_done_indexed(self) -> None:
+    async def test_wait_key_already_done_indexed(self, ts: TelescopeState) -> None:
         """Calling wait_key with a condition that is met must return (indexed version)."""
-        await self.ts.set_indexed('test_key', 'idx', 5)
-        await self.ts.wait_key('test_key', lambda v, t: v == {'idx': 5} and t is None)
+        await ts.set_indexed('test_key', 'idx', 5)
+        await ts.wait_key('test_key', lambda v, t: v == {'idx': 5} and t is None)
 
-    async def test_wait_key_timeout(self) -> None:
+    async def test_wait_key_timeout(self, ts: TelescopeState) -> None:
         """wait_key must time out in the given time if the condition is not met"""
-        with self.assertRaises(asyncio.TimeoutError):
-            with async_timeout.timeout(0.1):
-                await self.ts.wait_key('test_key')
-        with self.assertRaises(asyncio.TimeoutError):
-            with async_timeout.timeout(0.1):
+        with pytest.raises(asyncio.TimeoutError):
+            async with async_timeout.timeout(0.1):
```

Comment:
I see you added an `async` here. Was it needed before or has something else changed with `pytest-asyncio`?